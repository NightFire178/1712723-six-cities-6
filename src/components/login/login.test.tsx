import React, {useRef} from 'react'
import {render} from '@testing-library/react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import Login from "./login";
jest.mock('react', () => {
  const originReact = jest.requireActual('react');
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef,
  };
});
const appStateSelection = {
  isAuth:()=>({
    now:true
  })
}
jest.mock("../../redux/selectors/app-state", ()=> appStateSelection)
//eslint-disable-next-line
const handleSubmit = ()=> useRef()
const errors = ()=>[]
const register = useRef()
const useForm = ()=>({register, handleSubmit, errors})
jest.mock('react-hook-form', ()=>({useForm}))

//eslint-disable-next-line
jest.mock("../block/header/header", ()=>()=><div>Header</div>)

const thunkPostLogin =()=>true
jest.mock('../../hooks/use-thunk',() =>()=>({thunkPostLogin}))
const yupResolver = ()=>true
jest.mock('@hookform/resolvers/yup', ()=>({yupResolver}))
const yup = {
  object: ()=>({
    shape: ()=>true
  }),
  string: ()=>({
    required:()=>({
      email:()=>true,
      min: ()=>true
    })
  })
}
jest.mock(`yup`, ()=> (yup))

test(`login component block test`, ()=>{
  const history = createMemoryHistory()
  const {container} = render(
    <Router history={history}>
      <Login/>
    </Router>
  )
  expect(container).toMatchSnapshot()
})

