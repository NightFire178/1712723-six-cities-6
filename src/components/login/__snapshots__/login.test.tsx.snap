// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component block test 1`] = `<div />`;
